packages/relay-runtime/network/RelayObservable.js

```diff
@@ -477,6 +494,19 @@ function fromValue<T>(value: T): RelayObservable<T> {
   });
 }
 
+function fromAsyncIterator<T>(obj: AsyncIterator<T>): RelayObservable<T> {
+  return RelayObservable.create(sink => {
+    (async () => {
+      while (true) {
+        const {value, done} = await obj.next();
+        if (done) {break;}
+        sink.next(value);
+      }
+      sink.complete();
+    })().catch(sink.error);
+  });
```

Comment:
```suggestion
    return () => obj.return?.()
  });
```

The iterator should be returned when the RelayObservable is getting disposed of.